clients/src/main/java/org/apache/kafka/clients/producer/Partitioner.java

```diff
@@ -44,4 +44,14 @@
      */
     public void close();
 
+
+    /**
+     * Executes right before a new batch will be created. For example, if a sticky partitioner is used,
+     * this method can change the chosen sticky partition for the new batch. 
+     * @param topic The topic name
+     * @param cluster The current cluster metadata
+     * @param prevPartition The partition of the batch that was just completed
```

Comment:
This isn't 100% accurate, since a batch may not have been completed, right?

Maybe "prevPartition = the partition we previously selected"?